app.py

```diff
@@ -38,7 +38,48 @@ def mirror(name):
     }
     return create_response(data)
 
-# TODO: Implement the rest of the API here!
+@app.route('/users', methods = ['GET', 'POST'])
+def users():
+    if request.method == 'GET':
+        if request.args.get('team') is None:
+            data = {
+                'users': db.get('users')
+            }
+        else:
+            matched = [i for i in db.get('users') if i['team'] == request.args.get('team')]
+            data = {
+                'users': matched
+            }
+        return create_response(data)
+    elif request.method == 'POST':
+        input = request.get_json()
+        payload = {'name': input.get('name'),
+                   'age': input.get('age'),
+                   'team': input.get('team')}
+        return create_response(db.create('users', payload))
+
+@app.route('/users/<id>', methods = ['GET', 'PUT', 'DELETE'])
+def userById(id):
+    if request.method == 'GET':
+        if db.getById('users', int(id)) is None:
+            return create_response(None, 404, "User cannot be found")
+        else:
+            data = {
+                'user': db.getById('users', int(id))
+            }
+            return create_response(data)
+    elif request.method == 'PUT':
+        payload = {'name': request.form['name'], 'age': request.form['age'], 'team': request.form['team']}
+        if db.updateById('users', int(id), payload) is None:
+            return create_response(None, 404, "User cannot be found")
+        else:
+            return create_response(db.updateById('users', id, payload))
```

Comment:
`db.updateById` will actually update the database each time you call it. So here, you're actually updating it twice (once from the `if`, and a second time from your `return` when the function doesn't return `None`) with the same information except `id` will be a string instead of an integer